core/src/main/java/org/apache/carbondata/scan/scanner/impl/FilterScanner.java

```diff
@@ -151,7 +160,13 @@ private void fillScannedResult(BlocksChunkHolder blocksChunkHolder)
     for (int i = bitSet.nextSetBit(0); i >= 0; i = bitSet.nextSetBit(i + 1)) {
       indexes[index++] = i;
     }
-
+    // loading delete data cache in blockexecutioninfo instance
+    DeleteDeltaCacheLoaderIntf deleteCacheLoader =
+        new BlockletDeleteDeltaCacheLoader(scannedResult.getBlockletId(), blocksChunkHolder
+            .getDataBlock(), blockExecutionInfo.getAbsoluteTableIdentifier());
+    deleteCacheLoader.loadDeleteDeltaFileDataToCache();
+    scannedResult
```

Comment:
why we need to set delete delta cache in scan result, i think better to get the delete delta cache bitset and we can do And of filter scanner bit set here and set the resulting bit set to scan result so changes will be less and impact will also reduce